src/coreclr/gc/gc.cpp

```diff
@@ -2943,7 +2986,22 @@ void gc_heap::fire_pevents()
     settings.record (current_gc_data_global);
     current_gc_data_global->print();
 
-    FIRE_EVENT(GCGlobalHeapHistory_V3,
+#ifdef FEATURE_EVENT_TRACE
+    if (!informational_event_enabled_p) return;
+
+    uint32_t count_time_info = (settings.concurrent ? max_bgc_time_type : 
+                                (settings.compaction ? max_compact_time_type : max_sweep_time_type));
+
+    uint64_t* time_info = (settings.concurrent ? bgc_time_info : gc_time_info);
+    // We don't want to have to fire the time info as 64-bit integers as there's no need to
+    // so compress them down to 32-bit ones.
+    uint32_t* time_info_32 = (uint32_t*)time_info;
+    for (uint32_t i = 0; i < count_time_info; i++)
+    {
+        time_info_32[i] = ((time_info[i] > UINT32_MAX) ? UINT32_MAX : (uint32_t)time_info[i]);
```

Comment:
Nit: this pattern of a saturating conversion to 32 bits is repeated 5 times or so - perhaps it would be good to encapsulate in an inline method like "limit_time_to_uint32"?

Reply:
will do!